### ISSUE TITLE
Skip Lazy attributes in duplicate check

### ISSUE THREAD

**Issue by jfancher** [immutables/immutables#665]
Here's what I'm trying to accomplish:
````java
@Value.Style(get = {"get*", "internal*"})
@Value.Immutable
public abstract class Foo {
  // use hidden basic collection property to generate useful 'with' and builder methods
  abstract List<String> internalStrings();
  // but expose a collection wrapper that's nicer to use
  @Value.Lazy
  public MyStringList getStrings() { return MyStringList.from(internalStrings()); }
}
````

This generates an error because there are two attributes named 'strings' -- however, the generator doesn't need to create any methods for the lazy property. It only needs to override the getter, which must already be unique or else the abstract type wouldn't compile. It seems like the only thing that would conflict is the private field, which could easily be changed for lazy attributes.

I think the final result I want could be done more directly with `@Encoding`, but that seems like a lot of extra code for this simple case that _almost_ works as is.

I admit this is probably a pretty obscure usage, and will understand if you don't want to do it to keep things simpler, but I thought it wouldn't hurt to ask if there was something that could be done to enable this.

**Comment by elucash** [comment]
When I've started to investigate if this can be easily implemented, i.e. relaxing duplicate check for lazy attributes, I found out that major obstacle is that simplified property name is used to create backing field. If check is relaxed than fields will have potential for conflict. It's possible to special case lazy attribute names (by including prefixes like `lazy$` etc) but this will uglify generated code and indeed will introduce more complexity (by special casing). Among workarounds I can suggest removing `"get*"` here. I'll see, maybe it can be mitigated by allowing per-method specification of attribute name, such functionality not yet exists, but it can be folded into new functionality for customizing attribute/fields (including field annotations etc).
